### PR TITLE
go2tv: init at 1.13.0

### PR DESCRIPTION
**Pull request by gdamjan**
Cast media files to UPnP/DLNA Media Renderers and Smart TVs.
https://github.com/alexballas/go2tv

provides 2 packages:
- go2tv the GUI version (based on go-gl/glfw)
- go2tv-lite just a cli version

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [x] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/video/go2tv/default.nix [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+  ldflags = [
+    "-s" "-w"
+    "-linkmode=external"
+  ];
+
+  meta = with lib; {
+    description = "Cast media files to UPnP/DLNA Media Renderers and Smart TVs";
+    homepage = "https://github.com/alexballas/go2tv";
+    license = licenses.mit;
```
Comment (SuperSandro2000) [inline pkgs/applications/video/go2tv/default.nix:39]: Missing meta.maintainers

File: pkgs/applications/video/go2tv/default.nix [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+  doCheck = false;
+
+  buildInputs = [
+    xorg.libX11.dev
+    xorg.libXcursor.dev
+    xorg.libXrandr.dev
+    xorg.libXinerama.dev
+    xorg.libXi.dev
+    xorg.libXext.dev
```
Comment (SuperSandro2000) [inline pkgs/applications/video/go2tv/default.nix:24]: dev output should be choosen automatically

Comment (gdamjan) [reply pkgs/applications/video/go2tv/default.nix]: removed `.dev`

File: pkgs/applications/video/go2tv/default.nix (hunk 1 above)
Comment (gdamjan) [reply pkgs/applications/video/go2tv/default.nix:39]: added myself

**Comment by gdamjan** [comment]
updated to create 2 separate packages, one just cli (go2tv-lite), the other a gui version (based on go-gl/glfw).


I'm still not sure what's the proper way to provide the optional dependency of ffmpeg. go2tv just runs the executable if it's available, to do transcoding.

**Comment by SuperSandro2000** [comment]
> I'm still not sure what's the proper way to provide the optional dependency of ffmpeg. go2tv just runs the executable if it's available, to do transcoding.

If we don't want to add it to a wrapper all the time there is none. Optional wrappers just to populate PATH are overkill.

File: pkgs/applications/video/go2tv/default.nix [hunk 3]
```diff
@@ -0,0 +1,44 @@
+{ lib, buildGoModule, fetchFromGitHub, xorg, libglvnd, pkg-config, withGui ? true }:
+
+buildGoModule rec {
+  pname = if withGui then "go2tv" else "go2tv-lite";
```
Comment (SuperSandro2000) [inline pkgs/applications/video/go2tv/default.nix:4]:
```suggestion
  pname = "go2tv" + lib.optionalString (!withGui) "-lite";
```

File: pkgs/applications/video/go2tv/default.nix [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+    repo = "go2tv";
+    rev = "v${version}";
+    sha256 = "sha256-ZHKfBKOX3/kVR6Nc+jSmLgfmpihc6QMb6NvTFlsBr5E=";
+  };
+
+  vendorSha256 = "sha256-msXfXFWXyZeT6zrRPZkBV7PEyPqYkx+JlpTWUwgFavI=";
+
+  doCheck = false;
+
+  buildInputs = [
```
Comment (SuperSandro2000) [inline pkgs/applications/video/go2tv/default.nix:18]: The withGui boolean must be used somewhere.

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

**Comment by gdamjan** [comment]
> If we don't want to add it to a wrapper all the time there is none. Optional wrappers just to populate PATH are overkill.

maybe I should add a `withFfmpeg = false;` argument, so that if anyone wants to build manually they can do it?

File: pkgs/applications/video/go2tv/default.nix (hunk 4 above)
Comment (gdamjan) [reply pkgs/applications/video/go2tv/default.nix:22]: I've improved it somewhat now, does it make more sense?

File: pkgs/applications/video/go2tv/default.nix [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+  ];
+
+  nativeBuildInputs = [ pkg-config ];
+
+  ldflags = [
+    "-s" "-w"
+    "-linkmode=external"
+  ];
+
+  subPackages = subPackage;
```
Comment (SuperSandro2000) [inline pkgs/applications/video/go2tv/default.nix:40]:
```suggestion
  # conditionally build with GUI or not
  subPackages = [ "cmd/${pname}" ];
```

File: pkgs/applications/video/go2tv/default.nix (hunk 4 above)
Comment (SuperSandro2000) [reply pkgs/applications/video/go2tv/default.nix:22]: Ah, that wasn't obvious to me. We don't need to add extra code for that but a nice short comment is enough.

File: pkgs/applications/video/go2tv/default.nix [hunk 6]
```diff
@@ -0,0 +1,48 @@
+{ lib, buildGoModule, fetchFromGitHub, xorg, libglvnd, pkg-config, withGui ? true }:
+
+let
+  cmdName = "go2tv" + lib.optionalString (!withGui) "-lite";
+  subPackage = "cmd/${cmdName}";
+in
+buildGoModule rec {
+  pname = cmdName;
```
Comment (SuperSandro2000) [inline pkgs/applications/video/go2tv/default.nix:8]:
```suggestion
buildGoModule rec {
  pname = "go2tv" + lib.optionalString (!withGui) "-lite";
```

File: maintainers/maintainer-list.nix [hunk 7]
```diff
@@ -4915,6 +4915,13 @@
     githubId = 37017396;
     name = "gbtb";
   };
+  gdamjan = {
```
Comment (SuperSandro2000) [inline maintainers/maintainer-list.nix:4918]: Please put this into a separate commit with the message ``maintainers: add NAME`` and sort it alphabetically.

File: pkgs/applications/video/go2tv/default.nix (hunk 4 above)
Comment (gdamjan) [reply pkgs/applications/video/go2tv/default.nix:18]: I could also make buildInputs an empty list for withGui=false perhaps?

(doesn't change the -lite binary and its dependencies though)

File: pkgs/applications/video/go2tv/default.nix [hunk 8]
```diff
@@ -0,0 +1,45 @@
...
+  buildInputs = [
+    xorg.libX11
+    xorg.libXcursor
+    xorg.libXrandr
+    xorg.libXinerama
+    xorg.libXi
+    xorg.libXext
+    xorg.libXxf86vm
+    libglvnd
+  ];
```
Comment (SuperSandro2000) [inline pkgs/applications/video/go2tv/default.nix:27]:
```suggestion
  ] ++ lib.optionals stdenv.isDarwin [ Carbon ];
```

File: pkgs/applications/video/go2tv/default.nix [hunk 9]
```diff
@@ -0,0 +1,45 @@
+{ lib, buildGoModule, fetchFromGitHub, xorg, libglvnd, pkg-config, withGui ? true }:
```
Comment (SuperSandro2000) [inline pkgs/applications/video/go2tv/default.nix:1]:
```suggestion
{ lib
, buildGoModule
, fetchFromGitHub
, Carbon
, xorg
, libglvnd
, pkg-config
, withGui ? true
}:
```

File: pkgs/top-level/all-packages.nix [hunk 10]
```diff
@@ -2690,6 +2690,9 @@ with pkgs;
 
   gmnitohtml = callPackage ../applications/misc/gmnitohtml { };
 
+  go2tv = callPackage ../applications/video/go2tv { };
```
Comment (SuperSandro2000) [inline pkgs/top-level/all-packages.nix:2693]:
```suggestion
  go2tv = callPackage ../applications/video/go2tv {
    inherit (darwin.apple_sdk.frameworks) Carbon;
  };
```

File: pkgs/top-level/all-packages.nix [hunk 11]
```diff
@@ -2690,6 +2690,9 @@ with pkgs;
...
+  go2tv-lite = callPackage ../applications/video/go2tv { withGui = false; };
```
Comment (SuperSandro2000) [inline pkgs/top-level/all-packages.nix:2694]:
```suggestion
  go2tv-lite = go2tv.override { withGui = false; };
```

File: pkgs/applications/video/go2tv/default.nix [hunk 12]
```diff
@@ -0,0 +1,45 @@
...
+    xorg.libXrandr
+    xorg.libXinerama
+    xorg.libXi
+    xorg.libXext
+    xorg.libXxf86vm
+    libglvnd
+  ];
+
+  nativeBuildInputs = [ pkg-config ];
+
```
Comment (SuperSandro2000) [inline pkgs/applications/video/go2tv/default.nix:30]:
```suggestion
```

File: pkgs/applications/video/go2tv/default.nix [hunk 13]
```diff
@@ -0,0 +1,45 @@
...
+    owner = "alexballas";
+    repo = "go2tv";
+    rev = "v${version}";
+    sha256 = "sha256-ZHKfBKOX3/kVR6Nc+jSmLgfmpihc6QMb6NvTFlsBr5E=";
+  };
+
+  vendorSha256 = "sha256-msXfXFWXyZeT6zrRPZkBV7PEyPqYkx+JlpTWUwgFavI=";
+
+  doCheck = false;
+
```
Comment (SuperSandro2000) [inline pkgs/applications/video/go2tv/default.nix:17]:
```suggestion

  nativeBuildInputs = [ pkg-config ];

```

**Comment by SuperSandro2000** [comment]
@ofborg build go2tv

**Comment by SuperSandro2000** [comment]
darwin now fails with 

```
# fyne.io/fyne/v2/internal/driver/glfw
menu_darwin.m:10:1: error: unknown type name 'NSControlStateValue'; did you mean 'NSCellStateValue'?
/nix/store/yydafvwas3hia0rl0f30mmhv2kh5lazx-apple-framework-AppKit/Library/Frameworks/AppKit.framework/Headers/NSCell.h:73:19: note: 'NSCellStateValue' declared here
menu_darwin.m:10:32: error: use of undeclared identifier 'NSControlStateValueOn'
menu_darwin.m:11:1: error: unknown type name 'NSControlStateValue'; did you mean 'NSCellStateValue'?
/nix/store/yydafvwas3hia0rl0f30mmhv2kh5lazx-apple-framework-AppKit/Library/Frameworks/AppKit.framework/Headers/NSCell.h:73:19: note: 'NSCellStateValue' declared here
menu_darwin.m:11:33: error: use of undeclared identifier 'NSControlStateValueOff'
```

**Comment by gdamjan** [comment]
thanks @SuperSandro2000 for testing this, I don't have a way to test on darwin.

I see their Github Action for macos releases uses something called "fyne", checking it out now.

**Comment by gdamjan** [comment]
maybe @alexballas has an idea why the build fails


only thing I've found is:
```
2017-12-22  Wenson Hsieh  <wenson_hsieh@apple.com>

        Unreviewed, try to fix the Sierra build after r226277.

        The macOS 10.12 SDK does not know about NSControlStateValue and some types of NSLevelIndicatorStyles, so these
        need to be declared in a separate section in AppKitCompatibilityDeclarations.h (rather than in the < macOS 10.12
        SDK #ifdef).

        * wtf/mac/AppKitCompatibilityDeclarations.h:```

**Comment by SuperSandro2000** [comment]
That sounds promising

**Comment by SuperSandro2000** [comment]
Lets try it with the newer sdk and if that does not work we mark it broken.

**Comment by gdamjan** [comment]
```
app_darwin.m:6:9: fatal error: 'UserNotifications/UserNotifications.h' file not found
```

I've added PushKit too

**Comment by SuperSandro2000** [comment]
it builds on x84_64-darwin 🎉

**Comment by gdamjan** [comment]
Thanks a lot :)
